gobblin-metrics/src/main/java/gobblin/metrics/KafkaReporter.java

```diff
@@ -0,0 +1,426 @@
+      return self();
+    }
+
+    /**
+     * Convert rates to the given time unit.
+     *
+     * @param rateUnit a unit of time
+     * @return {@code this}
+     */
+    public T convertRatesTo(TimeUnit rateUnit) {
+      this.rateUnit = rateUnit;
+      return self();
+    }
+
+    /**
+     * Convert durations to the given time unit.
+     *
+     * @param durationUnit a unit of time
+     * @return {@code this}
+     */
+    public T convertDurationsTo(TimeUnit durationUnit) {
+      this.durationUnit = durationUnit;
+      return self();
+    }
+
+    /**
+     * Add tags
+     * @param tags
+     * @return {@code this}
+     */
+    public T withTags(Map<String, String> tags) {
+      this.tags.putAll(tags);
+      return self();
+    }
+
+    /**
+     * Add tags
+     * @param tags List of {@link gobblin.metrics.Tag}
+     * @return {@code this}
+     */
+    public T withTags(List<Tag<?>> tags) {
+      for(Tag<?> tag : tags) {
+        this.tags.put(tag.getKey(), tag.getValue().toString());
+      }
+      return self();
+    }
+
+    /**
+     * Add tag
+     * @param key tag key
+     * @param value tag value
+     * @return {@code this}
+     */
+    public T withTag(String key, String value) {
+      this.tags.put(key, value);
+      return self();
+    }
+
+    /**
+     * Builds and returns {@link gobblin.metrics.KafkaReporter}
+     *
+     * @param brokers string of Kafka brokers
+     * @param topic topic to send metrics to
+     * @return KafkaReporter
+     */
+    public KafkaReporter build(String brokers, String topic) {
+      this.brokers = brokers;
+      this.topic = topic;
+      return new KafkaReporter(this);
+    }
+
+  }
+
+  /**
+   * Class to contain a single metric
+   */
+  public static class Metric {
+    public String name;
+    public Object value;
+    public Map<String, String> tags;
+    public long timestamp;
+  }
+
+  /**
+   * Builds a metric name by appending elements of the path to the base name
+   * @param name base name of metric
+   * @param path elements to append to the base name
+   * @return generated name
+   */
+  public static String makeName(String name, String... path) {
+    final StringBuilder sb = new StringBuilder(name);
+    for (String part: path) {
+      sb.append('.').append(part);
+    }
+    return sb.toString();
+  }
+
+  protected KafkaReporter(Builder<?> builder) {
+    super(builder.registry, builder.name, builder.filter,
+        builder.rateUnit, builder.durationUnit);
+
+    lastSerializeExceptionTime = 0;
+
+    this.tags = builder.tags;
+
+    Properties props = new Properties();
+
+    props.put("metadata.broker.list", builder.brokers);
+    props.put("serializer.class", "kafka.serializer.DefaultEncoder");
+    props.put("request.required.acks", "1");
+
+    config = new ProducerConfig(props);
+    producer = new Producer<String, byte[]>(config);
+    this.topic = builder.topic;
+
+  }
+
+  @Override
+  public void close() {
+    super.close();
```

Comment:
Super.close may throw IOException and prevents producer.close() from being called. If producer.close() is from Closeable, please use a Closer.